### PR TITLE
Events for calculator interaction

### PR DESCRIPTION
**Pull request by tylher**
- [x] Use state to control the input and output of the calculator
- [x] Make use of imported function in handling click events

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+/**
+ * Given a button name and a calculator data object, return an updated
+ * calculator data object.
+ *
+ * Calculator data object contains:
+ *   total:s      the running total
+ *   next:String       the next number to be operated on with the total
+ *   operation:String  +, -, etc.
```
Comment (Whoistolu) [inline src/logic/calculate.js]: - [x] [OPTIONAL] Kindly remove all unused code in order to have a clean code-base.

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    return {};
+  }
+
+  // Button must be an operation
+
+  // When the user presses an operation button without having entered
+  // a number first, do nothing.
+  // if (!obj.next && !obj.total) {
+  //   return {};
+  // }
```
Comment (Whoistolu) [inline src/logic/calculate.js]: - [x] [OPTIONAL] Kindly remove all unused code in order to have a clean code base.

**Review by Whoistolu — APPROVED** [review]
### Hi @tylher 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/calculate.js (hunk 1 above)
Comment (tylher) [reply src/logic/calculate.js]: Alright, thank you for this observation, I will make the necessary corrections

**Pull request merged**